chatGPT_API/Controller/StoryViewController.swift

```diff
@@ -0,0 +1,172 @@
+    
+    var currentTitle = ""
+    var currentOption = ""
+    
+    
+
+    override func viewDidLoad() {
+        super.viewDidLoad()
+        setup()
+        model.delegate = self
+        model.fetch(with: queryManager.queryOpening)
+        
+    }
+//triggered when any of the option is chosen
+    @IBAction func choiceMade(_ sender: UIButton) {
+        currentOption = sender.currentTitle!
+        UIView.animate(withDuration: 0.5, animations:{
+            self.historyBTN.alpha = 100
+        })
+        HistoryModel.shared.addData(title: storyTitle.text!, option: currentOption)
+        print(HistoryModel.shared.data)
+        hideLabel()
+        hideButton()
+        gifPlayer.stop()
+        loadingText.text = "Continuing your story"
+        gifPlayer.playGIF(named: "pusheen_continue", in: player)
+        DispatchQueue.main.asyncAfter(deadline: .now() + 0.25) {
+            // Call second function
+            self.showPlayer()
+        }
+        
+        model.fetch(with: queryManager.continueQuery(story: currentTitle, option: currentOption))
+    }
+    
+    @IBAction func choiceButtonPressed(_ sender: Any) {
+        self.performSegue(withIdentifier: "historyPopUp", sender: self)
+    }
+    
+}
+
+//MARK: - Story Delegate
+//delegate to update the story once response came to DomainModel
+extension StoryViewController: StoryManagerDelegate{
+    func didUpdateStory(story: String) {
+        let modeledStory = storyBrain.modelAStory(input: model.content)
+        if modeledStory?.title == "" || (modeledStory?.title.count)! < 15{
+            print("Title is empty. It's wrong and I'm trying not to change it")
+            model.fetch(with: queryManager.continueQuery(story: currentTitle, option: currentOption))
+        }else{
+            self.option1.setTitle(modeledStory?.choice1, for: .normal)
+            self.option1.sizeToFit()
+            self.option2.setTitle(modeledStory?.choice2, for: .normal)
+            self.option2.sizeToFit()
+            self.option3.setTitle(modeledStory?.choice3, for: .normal)
+            self.option3.sizeToFit()
+            storyAnimate(storyLabel: self.storyTitle, storyTitle: modeledStory!.title)
+        }
+    }
+}
+    
+//MARK: - Animate
+//Letter by letter animated text
+extension UILabel {
+    func animate(newText: String, characterDelay: TimeInterval, completion: (() -> Void)? = nil) {
+        DispatchQueue.main.async() {
+            self.text = ""
+            for (index, character) in newText.enumerated() {
+                DispatchQueue.main.asyncAfter(deadline: .now() + characterDelay * Double(index)) {
+                    self.text?.append(character)
+                    if index == newText.count - 1 { // check if this is the last character to be animated
+                    completion?() // call the completion handler if it's not nil
+                }
+                }
+            }
+        }
+    }
+}
+//MARK: - Animations
+extension StoryViewController{
```

Comment:
Nice work separating these into their own functions! Can add `private` before the extension so all the functions aren't accessible to the outside.